sgt-back-end/index.js

```diff
@@ -0,0 +1,240 @@
+const pg = require('pg');
+const express = require('express');
+const app = express();
+app.use(express.json());
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.get('/api/grades/:gradeId', (req, res, next) => {
+  const gradeId = Number(req.params.gradeId);
+  if (!Number.isInteger(gradeId) || gradeId <= 0) {
+    res.status(400).json({
+      error: '"gradeId" must be a positive integer'
+    });
+    return;
+  }
+  const sql = `
+    select "gradeId",
+           "name",
+           "course",
+           "score",
+           "createdAt"
+      from "grades"
+     where "gradeId" = $1
+  `;
+  const params = [gradeId];
+  db.query(sql, params)
+    .then(result => {
+      const grade = result.rows[0];
+      if (!grade) {
+        res.status(404).json({
+          error: `Cannot find grade with gradeId: ${gradeId}`
+        });
+      } else {
+
+        res.json(grade);
+      }
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.get('/api/grades', (req, res, next) => {
+  const sql = `
+    select "gradeId",
+           "name",
+           "course",
+           "score",
+           "createdAt"
+      from "grades"
+  `;
+  db.query(sql)
+    .then(result => {
+      const grade = result.rows;
+      res.status(200).json(grade);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.post('/api/grades', function (req, res) {
+  if (!req.body.name) {
+    const error = { error: 'Name is a required field' };
+    res.status(400).json(error);
+    return;
+  } else if (!req.body.course) {
+    const error = { error: 'Course is a required field' };
+    res.status(400).json(error);
+    return;
+  } else if (!req.body.score && req.body.score !== 0) {
+    const error = { error: 'Score is a required field' };
+    res.status(400).json(error);
+    return;
+  } else if (req.body.score < 0 || req.body.score > 100) {
+    const error = { error: 'Score is invalid' };
+    res.status(400).json(error);
+    return;
+  }
+  const body = req.body;
+  const empty = [];
+  for (const x in body) {
+    empty.push(body[x]);
+  }
```

Comment:
Clever idea but this is actually a bad idea, there could be extra stuff in the `body` object that you don't want inside your SQL params, plus this doesn't guarantee the items in the array will be in the correct order. So it is much better to explicitly build the params array to ensure you are adding the correct properties in the correct order.